cmd/libsnap-confine-private/mount-opt-test.c

```diff
@@ -224,8 +227,14 @@ static void test_sc_do_mount(void)
 	}
 	g_test_trap_subprocess(NULL, 0, 0);
 	g_test_trap_assert_failed();
-	g_test_trap_assert_stderr
-	    ("cannot perform operation: mount -t ext4 -o ro /foo /bar: Permission denied\n");
+	if (GPOINTER_TO_INT(snap_debug) == 0) {
+		g_test_trap_assert_stderr
+		    ("cannot perform operation: mount -t ext4 -o ro /foo /bar: Permission denied\n");
+	} else {
+		g_test_trap_assert_stderr
+		    ("DEBUG: performing operation: (disabled) use debug build to see details\n"
+		     "cannot perform operation: mount -t ext4 -o ro /foo /bar: Permission denied\n");
```

Comment:
Pardon my ignorance - but why do I see the real operation here still? Isn't the point of the non-debug build to hide this? Maybe a comment here that explains what is going on?

Reply:
We expect it to be visible regardless of debug build and SNAP_CONFINE_DEBUG setting as it's useful for figuring out what failed exactly. Previously, in non debug builds, you'd only see the error to contain `mount ..` if you ran without SNAP_CONFINE_DEBUG. When running with SNAP_CONFINE_DEBUG, you'd get `(disabled) use debug build to see details` instead.

Reply:
This was meant as a security precaution against running diagnostic code in production builds. If that code contains any errors it could be used to attack a setuid-root program.

Reply:
I'm probably a bit slow and/or missing something - but if we show the real operation, why do we show the "diabled use debug build" message at all?